### PR TITLE
Override region according to bucket location

### PR DESCRIPTION
**Pull request by aspacca**
Enhancement
## What does this PR do?
Allows to ingest logs from aws-s3 bucket listing input regardless of the region used in the aws profile

## Why is it important?
https://github.com/elastic/observability-test-plans/issues/232#issuecomment-910333796

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [ ]

## Related issues

-

### CODE REVIEW

File: x-pack/filebeat/input/awss3/input.go [hunk 1]
```diff
@@ -246,3 +255,19 @@ func getRegionFromQueueURL(queueURL string, endpoint string) (string, error) {
 	}
 	return "", fmt.Errorf("QueueURL is not in format: https://sqs.{REGION_ENDPOINT}.{ENDPOINT}/{ACCOUNT_NUMBER}/{QUEUE_NAME}")
 }
+
+func getRegionForBucketARN(ctx context.Context, s3Client *s3.Client, bucketARN string) (string, error) {
```
Comment (kaiyan-sheng) [inline x-pack/filebeat/input/awss3/input.go:259]: add some unit tests for this function? 😄

Comment (aspacca) [reply x-pack/filebeat/input/awss3/input.go:259]: I can add an integration test

Comment (kaiyan-sheng) [reply x-pack/filebeat/input/awss3/input.go:259]: oh yes my bad, thanks!

**Review by kaiyan-sheng — APPROVED** [review]

**Comment by kaiyan-sheng** [comment]
Sorry I forgot one thing: should we update the aws-s3 documentation to include a new permission for GetBucketLocation?

**Review by kaiyan-sheng — APPROVED** [review]

**Pull request merged**